File: src/test/java/com/github/nigelzor/hogs/DirectionTest.java
package com.github.nigelzor.hogs;

import static org.hamcrest.CoreMatchers.is;
import static org.junit.Assert.*;

import org.junit.Test;

public class DirectionTest {

	@Test
	public void testRotateNorth() {
		assertThat(Direction.NORTH.rotate(Rotation.ZERO_DEGREES), is(Direction.NORTH));
		assertThat(Direction.NORTH.rotate(Rotation.NINETY_DEGREES), is(Direction.EAST));
		assertThat(Direction.NORTH.rotate(Rotation.ONE_HUNDRED_EIGHTY_DEGREES), is(Direction.SOUTH));
		assertThat(Direction.NORTH.rotate(Rotation.TWO_HUNDRED_SEVENTY_DEGREES), is(Direction.WEST));
	}

	@Test
	public void testRotateSouth() {
		assertThat(Direction.SOUTH.rotate(Rotation.ZERO_DEGREES), is(Direction.SOUTH));
		assertThat(Direction.SOUTH.rotate(Rotation.NINETY_DEGREES), is(Direction.WEST));
		assertThat(Direction.SOUTH.rotate(Rotation.ONE_HUNDRED_EIGHTY_DEGREES), is(Direction.NORTH));
		assertThat(Direction.SOUTH.rotate(Rotation.TWO_HUNDRED_SEVENTY_DEGREES), is(Direction.EAST));
	}

}

File: src/test/kotlin/com/github/nigelzor/hogs/DirectionTest.kt
package com.github.nigelzor.hogs

import kotlin.test.assertEquals
import org.junit.Test

class DirectionTest {

	@Test fun testRotateNorth() {
		assertEquals(Direction.NORTH, Direction.NORTH.rotate(Rotation.ZERO_DEGREES))
		assertEquals(Direction.EAST, Direction.NORTH.rotate(Rotation.NINETY_DEGREES))
		assertEquals(Direction.SOUTH, Direction.NORTH.rotate(Rotation.ONE_HUNDRED_EIGHTY_DEGREES))
		assertEquals(Direction.WEST, Direction.NORTH.rotate(Rotation.TWO_HUNDRED_SEVENTY_DEGREES))
	}

	@Test fun testRotateSouth() {
		assertEquals(Direction.SOUTH, Direction.SOUTH.rotate(Rotation.ZERO_DEGREES))
		assertEquals(Direction.WEST, Direction.SOUTH.rotate(Rotation.NINETY_DEGREES))
		assertEquals(Direction.NORTH, Direction.SOUTH.rotate(Rotation.ONE_HUNDRED_EIGHTY_DEGREES))
		assertEquals(Direction.EAST, Direction.SOUTH.rotate(Rotation.TWO_HUNDRED_SEVENTY_DEGREES))
	}

}
